src/chapters/functions/intro.rst

```diff
@@ -0,0 +1,98 @@
+
+Each of the functions we have used works in the same way. Using the function's name, followed by parentheses, we can *call* the function, resulting in some action being carried out. Sometimes, as with ``console.log``, we can provide input data within the parenteses, which the function will use to carry out its action. 
+
+.. admonition:: Example
+
+   The function ``console.log`` carries out the action of printing the provided value(s). In this case, ``"Hello, World!"`` is printed to the console.
+
+   .. sourcecode:: js
+
+      console.log("Hello, World!");
+
+   **Output**
+
+   ::
+
+      Hello, World!   
+   
+In addition to providing input, we may also be given a value in return when calling a function. This is what happened when we have used the type conversion functions, for example. 
+
+.. admonition:: Example
+
+   Type conversion functions *return* a value, that can be used by the calling code. Often, we store the return value of a function in a variable.
+
+   .. sourcecode:: js
+   
+      let num = Number("42");
+      console.log("The variable num is of type", typeof num, "and has value", num);
+
+   **Output**
+
+   ::
+
+      The variable num is of type number and has value 42
+
+.. index:: encapsulation
+
+Functions are extremely powerful in that they allow us to repeat actions without writing out the full code needed to carry out that action each time. They also allow us to be removed from the details of what the function is actually doing. 
+
+When we want to print a message to the console using ``console.log``, we do not need to be concerned with details such as what the console is, or how a string can be displayed on the console. This behavior is wrapped up within the function itself. We reap the benefits of ``console.log`` without needing to know how it works. This is an example of a broader programming concept known as **encapsulation**. Encapsulation is the process of packaging up code in a way that allows it to be resused, with the person or program using the encapsulated code not having to be concerned with the details of how it works.
+
+.. index::
+   single: function; machine
+
+A commonly-used analogy for describing the concept of a function is that of a machine that takes input, carries out some action, and gives back some result. This is known as the **function machine** analogy.
+
+.. figure:: figures/function-machine.png
+   :alt: A "function machine," consisting of a box which takes inputs, and from which output emerges.
+
+   The function machine
+
+If we want to use a function, we must provide it some input (if needed). It then carries out an action using the input provided, and returns a result. What happens in the middle, when the function is doing its work, is "inside the machine" in a sense. If we know how a function should behave, we simply provide input, and receive output. The rest is up to the function machine itself.
```

Comment:
Rephrasing to reduce passive voice.  How about "... It carries out an action on the input and returns a result.  The action occurs within the function, or 'inside the machine'.  If we know the purpose of a function, we simply provide it input and receive the output.  The rest is up to the machine itself."

If you want to reinforce the black box idea, add "without worrying about how one produces the other" to the end of the second to last sentence.